### ISSUE TITLE
Graceful parsing of Colorants

### ISSUE THREAD

**Issue by tbreloff** [JuliaGraphics/Colors.jl#232]
This isn't ideal behavior:

```
julia> using Colors

julia> parse(Colorant, "red")
RGB{U8}(1.0,0.0,0.0)

julia> parse(Colorant, colorant"red")
ERROR: StackOverflowError:
 in parse at /home/tom/.julia/v0.4/Colors/src/parse.jl:136 (repeats 342 times)
```

Should there be a better signature in [parse](https://github.com/JuliaGraphics/Colors.jl/blob/master/src/parse.jl#L136):

```
Base.parse{C<:Colorant}(::Type{C}, desc::AbstractString) = convert(C, parse(Colorant, desc))::C
```

and/or should there be a "pass-through" option for Colorants:

```
Base.parse{C<:Colorant}(::Type{C}, c::Colorant) = convert(C, c)
```

**Comment by timholy** [comment]
FYI `colorant"red"` by itself (without the parse) is all you need. But a pass-through would be great, if you want to submit that.

**Comment by tbreloff** [comment]
Yes... The issue came up somewhere in Plots, of course I'm still looking for the bug since the full stack trace isn't printed due to the stack overflow. (Is that a bug in julia?)

I'll put together a PR for the signatures I mentioned.

**Comment by timholy** [comment]
Ugh, StackOverflows shouldn't happen even in cases of user error, but I often forget to watch out for this.

The key technique is probably to change only one argument type at a time. Comparing https://github.com/JuliaGraphics/Colors.jl/blob/830936435de28ec1989728cbbde36f335d760a73/src/parse.jl#L75 and https://github.com/JuliaGraphics/Colors.jl/blob/830936435de28ec1989728cbbde36f335d760a73/src/parse.jl#L136, I'd predict the StackOverflow should go away if you add `::AbstractString` to the declaration of `desc` in the latter. If this is true, would be good to change this.

**Comment by timholy** [comment]
Fixed by #233
